crates/ra_ide_api/src/diagnostics.rs

```diff
@@ -212,12 +235,144 @@ mod tests {
         assert_eq_text!(after, &actual);
     }
 
+    fn check_no_diagnostic_for_target_file(target_file: &str, fixture: &str) {
+        let (analysis, file_id, _) = fixture_with_target_file(fixture, target_file);
+        let diagnostics = analysis.diagnostics(file_id).unwrap();
+        assert_eq!(diagnostics.len(), 0);
+    }
+
     fn check_no_diagnostic(content: &str) {
         let (analysis, file_id) = single_file(content);
         let diagnostics = analysis.diagnostics(file_id).unwrap();
         assert_eq!(diagnostics.len(), 0);
     }
 
+    #[test]
+    fn test_wrap_return_type() {
+        let before = r#"
+            //- /main.rs
+            use std::{string::String, result::Result::{self, Ok, Err}};
+
+            fn div(x: i32, y: i32) -> Result<i32, String> {
+                if y == 0 {
+                    return Err("div by zero".into());
+                }
+                x / y
```

Comment:
Instead of using `target_file`, let's use an `<|>` marker here and `with_position` method? We might also check, just in case, that diagnostic's range touches the `<|>`.